packages/labs/context/src/lib/context-root.ts

```diff
@@ -5,27 +5,30 @@
  */
 
 import {Context} from './create-context.js';
-import {ContextRequest, ContextRequestEvent} from './context-request-event.js';
+import {ContextCallback, ContextRequestEvent} from './context-request-event.js';
 import {ContextProviderEvent} from './controllers/context-provider.js';
 
-type UnknownContextKey = Context<unknown, unknown>;
-
 /**
- * A context request, with associated source element, with all objects as weak references.
- */
-type PendingContextRequest = Omit<
-  ContextRequest<UnknownContextKey>,
-  'context' | 'subscribe'
-> & {element: HTMLElement};
-
-/**
- * A ContextRoot can be used to gather unsatisfied context requests and redispatch these
- * requests when new providers which satisfy matching context keys are available.
+ * A ContextRoot buffers unsatisfied context request events. It will redispatch
+ * these requests when new providers which satisfy matching contexts
+ * are available.
  */
 export class ContextRoot {
   private pendingContextRequests = new Map<
-    UnknownContextKey,
-    Set<PendingContextRequest>
+    Context<unknown, unknown>,
+    {
+      // The WeakMap lets us detect if we're seen an element/callback pair yet
```

Comment:
```suggestion
      // The WeakMap lets us detect if we're seen an element/callback pair yet,
      // without needing to iterate the `requests` array
```
The info is in `requests`, it'd just be inefficient to need to loop over the array & deref each time?